File: 86-partition-list/partition-list.py
class Solution:
    def partition(self, head, x):
        
        small_dummy = ListNode(0)
        big_dummy = ListNode(0)
        
        small = small_dummy
        big = big_dummy
        
        while head:
            
            if head.val < x:
                small.next = head
                small = small.next
            else:
                big.next = head
                big = big.next
            
            head = head.next
        
        big.next = None
        small.next = big_dummy.next
        
        return small_dummy.next

File: 86-partition-list/partition-list.java
class Solution {
    public ListNode partition(ListNode head, int x) {
        
        ListNode smallDummy = new ListNode(0);
        ListNode bigDummy = new ListNode(0);
        
        ListNode small = smallDummy;
        ListNode big = bigDummy;
        
        while(head != null){
            
            if(head.val < x){
                small.next = head;
                small = small.next;
            }
            else{
                big.next = head;
                big = big.next;
            }
            
            head = head.next;
        }
        
        big.next = null;
        small.next = bigDummy.next;
        
        return smallDummy.next;
    }
}